### PR TITLE
Community logo with ActiveStorage

### PR DESCRIPTION
**Pull request by cwant**
This is mostly done. Could use another set of eyes though.

Things to watch out for:

* ActiveStorage monkey patched
* GlobalID added to Community to allow ActiveStorage to attach things to it
* `remove_logo` virtual attribute in Community -- not sure how I feel about this, but makes the controller/form easy

Still needed:

* More tests

### CODE REVIEW

File: app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -1,3 +1,4 @@
+//= require communities
```
Comment (murny) [inline app/assets/stylesheets/application.scss]: Same commet here: https://github.com/ualbertalib/jupiter/pull/104/files#r137047688

Use Sass `@import` over sprockets `require`

More info here: https://github.com/rails/sass-rails#important-note

Also put this stuff after vendor imports ( so you can reuse their variables among other reasons)

File: app/assets/stylesheets/communities.scss [hunk 2]
```diff
@@ -0,0 +1,16 @@
+$no_logo: #888888;
```
Comment (murny) [inline app/assets/stylesheets/communities.scss]: If you do the above, you could use bootstraps "gray" color instead. Just use $gray everywhere. It's defined as 
`#868e96` which is pretty close to your `#888888`

File: app/assets/stylesheets/communities.scss [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+.community-logo-small {
+  height: 100px !important;
```
Comment (murny) [inline app/assets/stylesheets/communities.scss]: Do you need to use `!important`?  Should be used sparingly and only as a last resort. Chances are you don't need it

File: app/models/community.rb [hunk 4]
```diff
@@ -8,8 +11,30 @@ def member_collections
     Collection.where(community_id: id)
   end
 
+  def logo
+    @active_storage_attached_logo ||
+      (@active_storage_attached_logo = ActiveStorage::Attached::One.new(:logo, self))
```
Comment (murny) [inline app/models/community.rb]: couldn't this be rewritten as:

`@active_storage_attached_logo ||= ActiveStorage::Attached::One.new(:logo, self)`?

File: app/assets/stylesheets/communities.scss (hunk 2 above)
Comment (cwant) [reply app/assets/stylesheets/communities.scss]: Hrmm, just using `gray` works, but using `$gray` throws an error. Using `gray` gives `#808080`, which is also very close.

File: app/models/community.rb (hunk 4 above)
Comment (cwant) [reply app/models/community.rb]: Yeah, I guess so. I was basically replicating what was happening here:

https://github.com/rails/activestorage/blob/v0.1/lib/active_storage/attached/macros.rb#L4

File: app/assets/stylesheets/communities.scss (hunk 3 above)
Comment (cwant) [reply app/assets/stylesheets/communities.scss]: Yep, not needed.

File: app/assets/stylesheets/communities.scss (hunk 2 above)
Comment (murny) [reply app/assets/stylesheets/communities.scss]: You should be  able to use `$gray` now after sass-rails is being used when #108 gets merged :confused:

Comment (cwant) [reply app/assets/stylesheets/communities.scss]: Hrmm, merged in from master now that #108 is in, still $gray doesn't work (do I need to run a rake task or something?).

Comment (murny) [reply app/assets/stylesheets/communities.scss]: Mmm wouldn't worry about it. I'll revisit this once this is merged. Probably something funky happening.

Comment (murny) [reply app/assets/stylesheets/communities.scss]: Yeah looks like they changed it: https://github.com/twbs/bootstrap/blob/v4-dev/scss/_variables.scss#L11-L62

So you can use one of the $gray-### ones or use the methods provided for accessing the  variable maps like: `colors("gray")` or `grays(600)`

**Comment by cwant** [comment]
Model tests added.

**Comment by mbarnett** [comment]
Whoops, thought of one more thing. We'll need to display the logos on the Community index page and in the admin section ala https://jupiter.mybalsamiq.com/projects/jupiter/Admin%20-%20Manage%20Communities%20and%20Collections

**Comment by cwant** [comment]
Okay, updated the "Done when" for the issue. This should be pretty straightforward.

**Comment by mbarnett** [comment]
Thanks!

**Comment by cwant** [comment]
Branch updated for logo on index page, this is what it looks like now:

community-index-logo

(Clicking the logo or "no logo" takes user to community, same as clicking the title link for the community.)

**Comment by cwant** [comment]
... and I just noticed that the "Has No Logo" in the screenshot means that the translation wasn't found. Will fix.

**Comment by cwant** [comment]
Okay, here we go:

community-index-logo2

Please kindly press the "Approve" button now!

**Pull request merged**